lua/kulala/inlay/init.lua

```diff
@@ -44,4 +51,66 @@ M.show = function(t, linenr)
   })
 end
 
+-- Function to add virtual text to patterns like {{host}}
+local add_virtual_variable_text = function()
+  if (M.show_virtual_variable_text or false) == false then
+    return
+  end
+
+  local bufnr = vim.api.nvim_get_current_buf()
+  local lines = vim.api.nvim_buf_get_lines(bufnr, 0, -1, false)
+
+  local ns_id = vim.api.nvim_create_namespace(VV_NS_NAME)
+  vim.api.nvim_buf_clear_namespace(bufnr, ns_id, 0, -1)
+
+  local pattern = "{{(.-)}}"
+
+  local variables = vim.tbl_extend(
+    "force",
+    TS.get_document_variables(),
+    ENV_PARSER.get_env(),
+    DB.data.http_client_env_base["DEFAULT_HEADERS"]
+  )
+
+  for lineno, line in ipairs(lines) do
+    for start_idx, match in line:gmatch("()(" .. pattern .. ")()") do
+      local label = match:gsub("{{", ""):gsub("}}", "")
+      local value = variables[label]
+      if value ~= nil then
+        -- Calculate the position to place virtual text
+        local end_idx = start_idx + #match - 1
+
+        -- Add virtual text before the closing braces of the match
+        vim.api.nvim_buf_set_extmark(bufnr, ns_id, lineno - 1, end_idx - 2, {
+          virt_text = { { ":" .. value, "Comment" } }, -- You can change the highlight group "Comment" as needed
+          virt_text_pos = "inline",
+        })
+      end
+    end
+  end
+end
+
+M.clear_virtual_variable_text = function()
+  local ns_id = vim.api.nvim_create_namespace(VV_NS_NAME)
+  local bufnr = vim.api.nvim_get_current_buf()
+  vim.api.nvim_buf_clear_namespace(bufnr, ns_id, 0, -1)
+  vim.api.nvim_clear_autocmds({ group = VV_GROUP_NAME })
+end
+
+M.toggle_virtual_variable = function()
+  M.show_virtual_variable_text = not M.show_virtual_variable_text
+
+  if M.show_virtual_variable_text then
+    add_virtual_variable_text()
+    vim.api.nvim_create_augroup(VV_GROUP_NAME, { clear = true })
+    vim.api.nvim_create_autocmd({ "BufEnter", "TextChanged", "TextChangedI" }, {
+      group = VV_GROUP_NAME,
+      pattern = "*.http",
```

Comment:
I think 🤔 we might also include *.rest here

Reply:
has done